### DISCUSSION TITLE
Support all VisualElement properties

### DISCUSSION THREAD

**Discussion by mrlacey** [CommunityToolkit/Maui.Markup#140] [New Feature Discussions]
Following on from https://github.com/CommunityToolkit/Maui.Markup/discussions/138#discussioncomment-4099379

As a starting point for filling in some gaps, I'd like to add fluent helper methods for all remaining settable properties of VIsualElement.

This would be the following properties:

- AnchorX
- AnchorY
- Background
- BackgroundColor
- Clip
- FlowDirection
- Frame
- InputTransparent
- IsEnabled
- IsVisible
- Opacity
- Rotation
- RotationX
- RotationY
- ZIndex

I expect some of these will only be used rarely, but it feels better to aim for completion than guess what people may or may not want/need.

Have I missed anything?
Is there a reason not to implement any of them?

These all appear to be able to be implemented with simple fluent extension methods.
Beyond the implementation and tests, is anything else needed for these?

**Comment by TheCodeTraveler** [comment]
That's sounds awesome!

Even if this list misses one or two, it's an AMAZING addition 💯

**Reply by bijington** [reply]
The only other item that we want is documentation entries added to our docs repo but I'll happily write those if you like

**Comment by mrlacey** [comment]
_Should_ the samples app include examples of everything too?

**Reply by bijington** [reply]
Yes ideally. I'm not 100% sure everything is but it would be great if it could be.

**Reply by TheCodeTraveler** [reply]
Just FYI - `CommunityToolkit.Maui.Markup.Samples` is a bit different than `CommunityToolkit.Maui.Samples`:
- [`CommunityToolkit.Maui.Samples`](https://github.com/CommunityToolkit/Maui/tree/main/samples) is a showcase of each feature in its own unique page
- [`CommunityToolkit.Maui.Markup.Samples`](https://github.com/CommunityToolkit/Maui.Markup/tree/main/samples) is more of a real-world app that subtly showcases each feature included in `CommunityToolkit.Maui.Markup`

I'm not sure which format is better, but so far we've been able to find a home for each C# Markup Extension in the existing sample app! Plus, I like showcasing how a real-world app would be architected using only C# (no XAML), as things like `AppShell` and Dark/Light Theming are a bit different in C# vs XAML.

**Comment by CliffAgius** [comment]
This is a stale thread so needs checking what is left on @mrlacey list here before closing this discussion,

**Reply by mrlacey** [reply]
Everything on the original list was added in #141.
With the exception of `Frame` which wasn't implemented but I don't recall why.
There are other properties of VisualElement that don't have specific fluent methods but I assume that as no one has brought them up in the last year, they're used so infrequently it doesn't matter. I'm sure they can be added as and when needed in the future.

Vote to close
